model/sharing/oauth.go

```diff
@@ -471,6 +472,43 @@ func (s *Sharing) SendAnswer(inst *instance.Instance, state string) error {
 	s.Credentials[0].Client = creds.Client
 	s.Active = true
 	s.Initial = s.NbFiles > 0
+
+	options := config.GetConfig().Sharing.OptionsForContext(inst.ContextName)
+	// Mark the sender's contact as trusted since we accepted their sharing
+	if len(s.Members) > 0 && s.Members[0].Email != "" && *options.AutoAcceptTrustedContacts {
```

Comment:
nit: putting `*options.AutoAcceptTrustedContacts` first makes it easier to see that this block will only be used in this situation.